### ISSUE TITLE
Chinese Search language problem

### ISSUE THREAD

**Issue by extremelyonline** [searxng/searxng#1321]
**Version of SearXNG**
I am using a public instance, all instances behave the same

**What happened?**
It seems that most search results under search language zh-HK 🇭🇰 are from China (in Simplified Chinese and not specific to the Hong Kong region), whereas the results under zh-TW 🇹🇼 are actually more accurate for HK users.

**How To Reproduce**

For instance, to look up "https://www.price.com.hk/" with the search term "price+hk", the first result under zh-HK is a site from China, whereas the first result under Taiwan is (closer to) the correct result.

**Expected behavior**
Search result should be from Hong Kong region when search language is set to zh-HK

**Screenshots & Logs**
![173271627-091c4b2d-e371-4f80-8f9e-87bfc90fa45f](https://user-images.githubusercontent.com/100532901/173472663-c448f0f0-7162-41d1-8477-fd5edcd9d96b.png)
![173271834-e5c3087a-f87b-476e-ae17-668059edbe3f](https://user-images.githubusercontent.com/100532901/173472664-68875e60-15de-4a76-a70d-28b482f895e7.png)

**Additional context**
No sure if this is a SearXNG problem or that of the engines

**Comment by return42** [comment]
> I am using a public instance, all instances behave the same

In my instance I test a new implementation of the region & language handling in SearXNG.

  https://darmarit.org/searx/

Can you please use my instance (in your daily searches) and report here if this instance gives you slightly better results ...

TLDR;

My instance is based on my [fork branch](https://github.com/searxng/searxng/compare/master...return42:fork) in which I test new stuff .. ATM I test a rework of the language & region handling of the engines .. (replaces the *language-centric* view of the implementations by a more *"language / region"* view, by example read this commit message https://github.com/searxng/searxng/commit/ba03f64bd118a7995f163d873576f1e144aa1b6e)

This rework needs to modify all engines that do support languages and I have fixed only a few engines right now in my branch .. but I expect slightly better results already now.

----

FYI: I mark related issues with the label [(locale & region)](https://github.com/searxng/searxng/labels/locale%20%26%20region)

**Comment by extremelyonline** [comment]
Thank you so much. So far it does seem more accurate.

Also just to inform you that the dictionary/define results are currently in Simplified Chinese under all the Chinese language/regions.

**Comment by return42** [comment]
> Also just to inform you that the dictionary/define results are currently in Simplified Chinese under all the Chinese language/regions.

Thanks for pointing out .. yes I know this is something that still needs to be done .. the work in my branch is still at the beginning .. it needs a lot of more work (the requests of all engines needs to be fixed).

I am happy to have some testers .. I will report any progress .. thanks a lot for your feedback :+1:
